data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/http/WebClientCloudEventSender.java

```diff
@@ -129,4 +134,12 @@ private void logError(final CloudEvent event, Throwable ex) {
     this.client.close();
     return Future.succeededFuture();
   }
+
+  static boolean isRetryableStatusCode(final int statusCode) {
+    // From https://github.com/knative/specs/blob/c348f501de9eb998b4fd010c54d9127033ee41be/specs/eventing/data-plane.md#event-acknowledgement-and-delivery-retry
+    return statusCode >= 500 || // Generic error
+      statusCode == 404 || // Endpoint does not exist
+      statusCode == 409 || // Conflict / Processing in progress
+      statusCode == 429;   // Too Many Requests / Overloaded
```

Comment:
for completeness we have `-1`  there as well, in upstream ...  

I guess that's due to a Golang thingy/bug... but just wanted to mention it here